In [1]:
#importing all the package you need
import pandas as pd
# Import the linear regression class
from sklearn.linear_model import LinearRegression
# Sklearn also has a helper that makes it easy to do cross validation
from sklearn.cross_validation import KFold
#https://www.dataquest.io/mission/74/getting-started-with-kaggle


C:\Users\Aska\Anaconda2\lib\site-packages\sklearn\cross_validation.py:44: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)


# Titanic Project

#Data Cleansing

In [2]:
# We can use the pandas library in python to read in the csv file.
# This creates a pandas dataframe and assigns it to the titanic variable.
titanic = pd.read_csv("train.csv")

# Print the first 5 rows of the dataframe.
print titanic.head(5)

#print titanic.describe()

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [3]:
#describe it
print titanic.describe(),'\n'
print titanic.dtypes


       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200   

PassengerId      int64
Survived         int64
Pclass           int64
Name      

In [4]:
#But There is missing values on "Age" columns. Because count of data point on "Age" columns is less than the other columns.
#we fill the missing values with the median of data. We can use other method to fill the missing value

titanic["Age"] = titanic["Age"].fillna(titanic["Age"].median())


We have to either exclude our non-numeric columns 
when we train our algorithm (Name, Sex, Cabin, Embarked, and Ticket), 
or find a way to convert them to numeric columns.

We'll ignore the Ticket, Cabin, and Name columns. 
There isn't much information we can extract from there. 
Most of the values in the cabin column are missing (only 204 values out of 891 rows), 
and it likely isn't a particularly informative column in the first place. The Ticket and Name columns are unlikely 
to tell us much without some domain knowledge about what the ticket numbers mean, 
and about which names correlate with characteristics like large or rich families.

In [5]:
#Now we work on non-numeric data
#Which also has missing values


#Sex doesn't has missing value
#Replace all the occurences of male with the number 0 and female with 1.
titanic.loc[titanic["Sex"] == "male", "Sex"] = 0
titanic.loc[titanic["Sex"] == "female", "Sex"] = 1

#Embarked columns has missing value
print titanic["Embarked"].unique()
titanic.Embarked = titanic.Embarked.fillna('S') # because most likely the passenger is 'S' 
titanic.loc[titanic.Embarked == "S", "Embarked"] = 0
titanic.loc[titanic.Embarked == "C", "Embarked"] = 1
titanic.loc[titanic.Embarked == "Q", "Embarked"] = 2


#titanic.head(10)

['S' 'C' 'Q' nan]


#On to ML

In [6]:
titanic.columns

Index([u'PassengerId', u'Survived', u'Pclass', u'Name', u'Sex', u'Age',
       u'SibSp', u'Parch', u'Ticket', u'Fare', u'Cabin', u'Embarked'],
      dtype='object')

In [7]:
titanic.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  891.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.361582    0.523008   
std     257.353842    0.486592    0.836071   13.019697    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   22.000000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   35.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [8]:
titanic.isnan()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris   0  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...   1  38.0      1      0   
2                             Heikkinen, Miss. Laina   1  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)   1  35.0      1      0   
4                           Allen, Mr. William Henry   0  35.0      0      0   
5                                   Moran, Mr. James   0  28.0      0      0   
6                            McCarthy, Mr. Timothy J   0  54.0      0      0   
7                     Palsson, Master. Gosta Leonard   0   2.0      3      1   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)   1  27.0      0      2   
9                Nasser, Mrs. Nicholas (Adele Achem)   1  14.0      1      0   

             Ticket     Fare Cabin Embarked  
0         A/5 21171   7.2500   NaN        0  
1          PC 17599  71.2833   C85        1  
2  STON/O2. 3101282   7.9250   NaN        0  
3            113803  53.1000  C123        0  
4            373450   8.0500   NaN        0  
5            330877   8.4583   NaN        2  
6             17463  51.8625   E46        0  
7            349909  21.0750   NaN        0  
8            347742  11.1333   NaN        0  
9            237736  30.0708   NaN        1